m2m-oauth-server/pb/token.go

```diff
@@ -0,0 +1,190 @@
+package pb
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+var errTokenIsNil = errors.New("Token is nil")
+
+func (x *Token) Validate() error {
+	if x == nil {
+		return errTokenIsNil
+	}
+	if x.GetId() == "" {
+		return errors.New("Token.Id is empty")
+	}
+	if x.GetOwner() == "" {
+		return errors.New("Token.Owner is empty")
+	}
+	if x.GetClientId() == "" {
+		return errors.New("Token.ClientId is empty")
+	}
+	if x.GetIssuedAt() == 0 {
+		return errors.New("Token.Timestamp is empty")
+	}
+	return nil
+}
+
+func (x *Token) ToMap() (map[string]interface{}, error) {
```

Comment:
The methods below are for encoding to BSON right? 

There are similar pkg methods `UnmarshalProtoBSON`, `MarshalProtoBSON` and `ConvertStringValueToInt64` that I created when I was implementing snippet-service. I think they do almost the same stuff, maybe we can combine them into a single encode/decode-bson API

Reply:
implemented